register_class.go

```diff
@@ -54,26 +54,31 @@ If the Struct implements an OnRegister(Lifetime) method, it will
 be called on a temporary instance when the class is registered.
 */
 func Register[Struct gd.Extends[Parent], Parent gd.IsClass](godot Lifetime) {
-	var classType = reflect.TypeOf([0]Struct{}).Elem()
-	var superType = reflect.TypeOf([0]Parent{}).Elem()
+	classType := reflect.TypeFor[Struct]()
+	superType := reflect.TypeFor[Parent]()
 	if classType.Kind() != reflect.Struct || classType.Name() == "" {
 		panic("gdextension.RegisterClass: Class type must be a named struct")
 	}
 
 	// Support 'gd' tag for renaming the class within Godot.
-	var rename = classNameOf(classType)
-
-	var tool = false
-	switch {
-	case superType.Implements(reflect.TypeOf([0]interface{ AsScript() Script }{}).Elem()),
-		superType.Implements(reflect.TypeOf([0]interface{ AsEditorPlugin() EditorPlugin }{}).Elem()),
-		superType.Implements(reflect.TypeOf([0]interface{ AsScriptLanguage() ScriptLanguage }{}).Elem()),
-		classType.Implements(reflect.TypeOf([0]Tool{}).Elem()):
+	rename := classNameOf(classType)
+
+	var parent Parent
```

Comment:
I believe this is the only extra allocation. In my experience, reflection is a lot slower than a reasonably small allocation on the stack. If the parent types are typically very large, reflection may cost less than the allocation.